### PR TITLE
add new APM billing metrics to sf.org

### PR DESCRIPTION
**Pull request by brianashby-sfx**
- reorganize existing apm metrics
- add new apm metrics

### CODE REVIEW

**Review by cbasavaiah-splunk — APPROVED** [review]

File: signalfx-org-metrics/metrics.yaml [hunk 1]
```diff
@@ -162,69 +286,37 @@ sf.org.apm.numTroubleshootingMetricSets:
   Dimension(s): `orgId`
 
   Data resolution: 1 minute'
-  metric_type:
+  metric_type: gauge
```
Comment (gergas3) [inline signalfx-org-metrics/metrics.yaml:289]: this is a counter

File: signalfx-org-metrics/metrics.yaml [hunk 2]
```diff
@@ -1,143 +1,267 @@
...
-  title: sf.org.apm.grossSpansReceived
+  title: sf.org.apm.grossSpanBytesReceived
 
-sf.org.apm.grossSpansReceivedByToken:
-  brief: The number of spans SignalFx receives for a specific access token
-    before filtering and throttling.
-  description: 'The number of spans SignalFx receives for a specific access token
-    before filtering or throttling.
+sf.org.apm.numSpanBytesReceived:
+  brief: The number of bytes Splunk APM accepts for a span after decompression
```
Comment (gergas3) [inline signalfx-org-metrics/metrics.yaml:54]: `for a span` --> `from all ingested span data`
Not for _a_ span, but bytes of _all_ spans ingested

File: signalfx-org-metrics/metrics.yaml [hunk 3]
```diff
@@ -1,143 +1,267 @@
...
   metric_type: counter
-  title: sf.org.apm.grossSpansReceivedByToken
+  title: sf.org.apm.numSpanBytesReceived
 
-sf.org.apm.numSpansReceived:
-  brief: The number of spans SignalFx accepts after filtering and throttling.
-  description: 'The number of spans SignalFx accepts after filtering and throttling.
+sf.org.apm.grossSpanBytesReceivedByToken:
+  brief: The number of bytes Splunk APM receives for a specific access token
+    from spans after decompression but before filtering and throttling.
```
Comment (gergas3) [inline signalfx-org-metrics/metrics.yaml:67]: `from spans` is also a good expression to use also with `sf.org.apm.numSpanBytesReceived`

**Review by gergas3 — CHANGES_REQUESTED** [review]

File: signalfx-org-metrics/README.md.jinja [hunk 4]
```diff
@@ -48,6 +48,13 @@ The `sf.org.numDetectorsAborted` metric monitors the number of detectors that we
 
 For information about events, see [View Additional Data With Events](https://docs.signalfx.com/en/latest/detect-alert/events-intro.html).
 
+### Compare gross and num metric values
+
+Some metrics report a `gross` value and a `num` value. Compare the `gross` and `num` values of a metric to learn about how the system limits or filters data for whatever the metric represents.
+
+A `gross` metric reports the total number of MTS the system receives before any throttling or filtering kicks in.
```
Comment (mdubbyap) [inline signalfx-org-metrics/README.md.jinja:55]: these two lines are incorrectly using MTS. datapoints would be a better noun for those calls but these also exist for spans and rum Spans.

File: signalfx-org-metrics/metrics.yaml [hunk 5]
```diff
@@ -1,143 +1,267 @@
...
 
-sf.org.apm.grossSpanBytesReceived:
-  brief: The number of bytes SignalFx receives from spans after decompression
-    but before filtering and throttling.
-  description: 'The number of bytes SignalFx receives from spans after
-    decompression but before filtering and throttling.
+sf.org.apm.numSpansDroppedInvalid:
+  brief: The number of invalid spans Splunk APM receives.
+  description: 'The number of invalid spans Splunk APM receives. A span can be invalid
+    if there is no start time, trace ID, or service associated with it.
```
Comment (mdubbyap) [inline signalfx-org-metrics/metrics.yaml:93]: too many spans in trace is another reason we would increment invalid spans dropped.

File: signalfx-org-metrics/metrics.yaml [hunk 6]
```diff
@@ -1,143 +1,267 @@
...
+    Dimension(s): `orgId`
+
+    Data resolution: 1 second'
+  metric_type: counter
+  title: sf.org.apm.numSpansDroppedInvalid
+
+sf.org.apm.numSpansDroppedInvalidByToken:
+  brief: The number of invalid spans Splunk APM receives for a specific access token.
+  description: 'The number of invalid spans Splunk APM receives for a specific access token.
+    A span can be invalid if there is no start time, trace ID, or service associated with it.
```
Comment (mdubbyap) [inline signalfx-org-metrics/metrics.yaml:104]: same as above.

**Review by gergas3 — APPROVED** [review]

**Pull request merged**